app/services/green/green.ts

```diff
@@ -0,0 +1,514 @@
+import { PersistentStatefulService, InitAfter, Inject, ViewHandler, mutation } from 'services/core';
+import {
+  TDisplayPlatforms,
+  TGreenPlatformSettings,
+  GreenPlatformSettings,
+  TGreenDisplayType,
+  IGreenPlatformSetting,
+} from './green-data';
+import { ScenesService, SceneItem, IPartialSettings } from 'services/scenes';
+import {
+  TDisplayType,
+  VideoSettingsService,
+  IVideoSetting,
+  greenDisplayData,
+} from 'services/settings-v2';
+import { StreamingService } from 'services/streaming';
+import { SceneCollectionsService } from 'services/scene-collections';
+import { TPlatform } from 'services/platforms';
+import { ReorderNodesCommand, EPlaceType } from 'services/editor-commands/commands/reorder-nodes';
+import { Subject } from 'rxjs';
+import { TOutputOrientation } from 'services/restream';
+import { IVideoInfo } from 'obs-studio-node';
+
+interface IDisplayVideoSettings {
+  defaultDisplay: TDisplayType;
+  green: IVideoInfo;
+  activeDisplays: {
+    horizontal: boolean;
+    green: boolean;
+  };
+}
+interface IGreenServiceState {
+  displays: TDisplayType[];
+  platformSettings: TGreenPlatformSettings;
+  greenMode: boolean;
+  nodeMaps: { [display in TDisplayType as string]: Dictionary<string> };
+  sceneNodeMaps: { [sceneId: string]: Dictionary<string> };
+  videoSettings: IDisplayVideoSettings;
+}
+
+class GreenViews extends ViewHandler<IGreenServiceState> {
+  @Inject() private scenesService: ScenesService;
+  @Inject() private videoSettingsService: VideoSettingsService;
+  @Inject() private streamingService: StreamingService;
+
+  get greenMode(): boolean {
+    return this.activeDisplays.horizontal && this.activeDisplays.green;
+  }
+
+  get platformSettings() {
+    return this.state.platformSettings;
+  }
+
+  get platformSettingsList(): IGreenPlatformSetting[] {
+    return Object.values(this.state.platformSettings);
+  }
+
+  get hasGreenScenes() {
+    if (!this.state.nodeMaps) return false;
+    for (const display in this.state.nodeMaps) {
+      if (!this.state.nodeMaps.hasOwnProperty(display)) {
+        return false;
+      }
+    }
+    return true;
+  }
+
+  get showGreenDisplays() {
+    return this.greenMode && this.hasGreenScenes;
+  }
+
+  get hasGreenNodes() {
+    return this.state.sceneNodeMaps.hasOwnProperty(this.scenesService.views.activeSceneId);
+  }
+
+  get greenNodeIds(): string[] {
+    const activeSceneId = this.scenesService.views.activeSceneId;
+
+    if (!this.hasGreenNodes) return;
+
+    return Object.entries(this.state.sceneNodeMaps[activeSceneId]).map(
+      ([key, value]: [string, string]) => value,
+    );
```

Comment:
Seems like this could just be `Object.values(this.state.sceneNodeMaps[activeSceneId])`?